### PR TITLE
Configure etcd `extraArgs` from TKR during cluster creation/upgrade

### LINKED ISSUE #1959: Configure `etcd.local.extraArgs` by reading the TKR `kubeadmConfigSpec.etcd`
**Bug description**

As per https://groups.google.com/a/kubernetes.io/g/dev/c/B7gJs88XtQc/m/rSgNOzV2BwAJ?utm_medium=email&utm_source=footer
there is a data corruption bug in the etcd 3.5 series.

Community recommendations are:
- Do not upgrade to etcd 3.5
- If you are on etcd 3.5, then `--experimental-initial-corrupt-check` should be added to the args for etcd as a basic mitigation - this will prevent etcd from starting if data corruption has already occurred.

There are some changes required in the CLI to configure `extraArgs` as part of `etcd` config from TKR during cluster creation/upgrade.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
When cluster gets created/upgrade, set `KubeadmControlPlane.spec.kubeadmConfigSpec.etcd.extraArgs.experimental-initial-corrupt-check: true` in all provider templates.

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)
Management Cluster version: v1.5.x

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- As per https://groups.google.com/a/kubernetes.io/g/dev/c/B7gJs88XtQc/m/rSgNOzV2BwAJ?utm_medium=email&utm_source=footer
  there is a data corruption bug in the etcd 3.5 series.
- As part of mitigation, we need to add `experimental-initial-corrupt-check` when configuring etcd
  in the extraArgs.
- This change configures extraArgs for etcd by reading the values from TKR

### Which issue(s) this PR fixes

Fixes #1959

### Describe testing done for PR
1. Create vsphere management cluster. 
   - `tanzu management-cluster create -f cluster-config/vsphere/mc-1/mc-1.yaml -v 6`
1. Once the management cluster gets created successfully verify the KCP resource with etcd configuration.
   - As the management-cluster deployed is using etcd v3.5.x verify the `experimental-initial-corrupt-check` is set to `true`
   - `kubectl get kcp aws-mc-1-control-plane -n tkg-system -ojson | jq '.spec.kubeadmConfigSpec.clusterConfiguration.etcd.local.extraArgs."experimental-initial-corrupt-check"'` returns `"true"`
1. Create workload cluster with v1.21 tkr which uses etcd v3.4
   - This should not include `experimental-initial-corrupt-check` configuration as part of KCP as etcd v3.4 is used
1. (pending) Upgrade the workload cluster to v1.22 tkg which uses etcd v3.5
   - After the upgrade `experimental-initial-corrupt-check: true` should be set as part of KCP configuration

### Release note

```release-note
Configure etcd extraArgs from TKR during cluster creation/upgrade
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by anujc25** [comment]
//usebom:sandbox/v1.5.3-zshippable/401131456904088358/tkg-compatibility

File: pkg/v1/tkg/tkgconfigbom/bom_test.go [hunk 1]
```diff
@@ -246,6 +246,34 @@ var (
 					Expect(err).ToNot(HaveOccurred())
 					Expect(bomConfiguration).ToNot(BeNil())
 				})
+				It("Should return the correct etcd configuration", func() {
+					Expect(err).ToNot(HaveOccurred())
+					Expect(bomConfiguration.KubeadmConfigSpec.Etcd.Local.DataDir).To(Equal("/var/lib/etcd"))
+					Expect(bomConfiguration.KubeadmConfigSpec.Etcd.Local.ImageRepository).To(Equal("registry.tkg.vmware.run"))
+					Expect(bomConfiguration.KubeadmConfigSpec.Etcd.Local.ImageTag).To(Equal("v3.4.13_vmware.4"))
+					Expect(len(bomConfiguration.KubeadmConfigSpec.Etcd.Local.ExtraArgs)).To(Equal(1))
+					Expect(bomConfiguration.KubeadmConfigSpec.Etcd.Local.ExtraArgs["experimental-initial-corrupt-check"]).To(Equal("true"))
```
Comment (vuil) [inline pkg/v1/tkg/tkgconfigbom/bom_test.go:255]: know it is just unit tests, but is this reflecting the opposite of reality.
iow, we shouldn't be expecting boms using pre-3.5 etcd to include the experimental-initial-corrupt-check, right?

**Review by vuil — APPROVED** [review]
LGTM, nice fix and tests! Just had a few small comments.

File: pkg/v1/tkg/tkgconfigbom/bom_test.go (hunk 1 above)
Comment (anujc25) [reply pkg/v1/tkg/tkgconfigbom/bom_test.go]: Yes. Updated to use fake arguments.

**Review by vuil — APPROVED** [review]
lgtm, thanks!

**Pull request merged**